pkg/printers/internalversion/printers.go

```diff
@@ -1154,11 +1154,24 @@ func formatHosts(rules []networking.IngressRule) string {
 	return ret
 }
 
-func formatPorts(tls []networking.IngressTLS) string {
-	if len(tls) != 0 {
-		return "80, 443"
+func formatPorts(obj *networking.Ingress) string {
+	allowHTTP := true
+	if annotationValue, exists := obj.Annotations["kubernetes.io/ingress.allow-http"]; exists {
+		allowHTTP, _ = strconv.ParseBool(annotationValue)
 	}
-	return "80"
+
+	if len(obj.Spec.TLS) != 0 {
+		if allowHTTP {
+			return "80, 443"
+		}
+		return "443"
+	}
+
+	if allowHTTP {
+		return "80"
+	}
+
+	return ""
```

Comment:
This should never be reached, so I think we can simplify by just returning 80 and drop one condition above.

Reply:
Changed it